tests/by-util/test_ln.rs

```diff
@@ -937,14 +883,37 @@ fn test_ln_non_utf8_paths() {
 
 #[test]
 fn test_ln_hard_link_dir() {
-    let scene = TestScenario::new(util_name!());
-    let at = &scene.fixtures;
-
+    let (at, mut ucmd) = at_and_ucmd!();
     at.mkdir("dir");
-
-    scene
-        .ucmd()
-        .args(&["dir", "dir_link"])
+    ucmd.args(&["dir", "dir_link"])
         .fails()
         .stderr_contains("hard link not allowed for directory");
 }
+
+#[test]
+fn test_ln_extra_operand() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    at.touch("a");
+    at.touch("b");
+    at.touch("c");
+    ucmd.args(&["-T", "a", "b", "c"])
+        .fails_with_code(1)
+        .stderr_contains("extra operand c")
+        .stderr_contains("--help");
+}
+
+#[cfg(target_os = "linux")]
+#[test]
+fn test_ln_cannot_stat_non_utf8() {
```

Comment:
I think this test is not correct. It matches the behavior of our implementation, however, if I do the steps manually with GNU `ln` (v9.9), I don't get an error:
```
$ mkdir target
$ touch $'file_\xFF'
$ ln -s file_�  target
$ echo $?
0
$ ls -l target/
total 0
lrwxrwxrwx 1 dho dho 6 Dec 11 10:21 'file_'$'\377' -> 'file_'$'\377'
```

Reply:
I think I was able to find the source of it. My understanding is that there was a to_str() that was was checked if None and that would only happen with non utf-8 filenames, by switching it to just calling filename directly, there's no optional. I made a commit that implements the fix for this too